tower-mock/src/lib.rs

```diff
@@ -307,22 +327,19 @@ impl<T, E> Respond<T, E> {
 
 // ===== impl ResponseFuture =====
 
-impl<T, E> Future for ResponseFuture<T, E> {
+impl<T, E> Future for ResponseFuture<T, E>
+where
+    E: Into<Error>,
+{
     type Item = T;
-    type Error = Error<E>;
+    type Error = Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        match self.rx {
-            Error::Other(ref mut rx) => {
-                match rx.poll() {
-                    Ok(Async::Ready(Ok(v))) => Ok(v.into()),
-                    Ok(Async::Ready(Err(e))) => Err(Error::Other(e)),
-                    Ok(Async::NotReady) => Ok(Async::NotReady),
-                    Err(_) => Err(Error::Closed),
-                }
-            }
-            Error::NoCapacity => Err(Error::NoCapacity),
-            Error::Closed => Err(Error::Closed),
+        match self.rx.poll() {
+            Ok(Async::Ready(Ok(v))) => Ok(v.into()),
+            Ok(Async::Ready(Err(e))) => Err(e.into()),
+            Ok(Async::NotReady) => Ok(Async::NotReady),
+            Err(e) => Err(e.into()),
```

Comment:
Again, the question mark operator is probably our friend here, I think this match could be something like
```rust
match self.rx.poll()? {
    Async::Ready(ready) => Ok(ready?),
    Async::NotReady => Ok(Async::NotReady),
}
```

Reply:
True, yes—I'll switch it over to `?`.